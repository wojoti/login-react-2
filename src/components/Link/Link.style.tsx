import styled from "styled-components";
type Props = {
  decoration?: string;
  color?: string;
};
export const Wrapper = styled.a<Props>`
  text-decoration: ${(props) => props.decoration || "underline"};
  color: ${(props) => props.color || "inherit"};
  font-size: 14px;
  &:link,
  &:visited {
    color: ${(props) => props.color || "inherit"};
  }
`;
