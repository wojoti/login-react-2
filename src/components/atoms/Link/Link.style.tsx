import styled from "styled-components";

interface Props {
  decoration?: string;
  color?: string;
}
const StyledLink = styled.button<Props>`
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  text-decoration: ${(props) => props.decoration || "underline"};
  color: ${(props) => props.color || "inherit"};
  font-size: 14px;

  background: none !important;
  border: none;
  padding: 0 !important;
  cursor: pointer;
`;
export default StyledLink;
