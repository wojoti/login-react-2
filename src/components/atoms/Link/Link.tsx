import StyledLink from "./Link.style";

export interface LinkProps {
  children: React.ReactNode;
  decoration?: string;
  color?: string;
  testId?: string;
  onClick: () => void;
}

function Link({ children, decoration, color, onClick, testId }: LinkProps) {
  return (
    <StyledLink
      type="button"
      decoration={decoration}
      color={color}
      data-testid={testId}
      onClick={onClick}
    >
      {children}
    </StyledLink>
  );
}
export default Link;
