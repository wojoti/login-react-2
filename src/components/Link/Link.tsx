import React, { ReactNode } from "react";
import { Wrapper } from "./Link.style";

type Props = {
  children?: ReactNode;
  decoration?: string;
  goto: string;
  color?: string;
};

const Link = (props: Props) => {
  return (
    <Wrapper
      decoration={props.decoration}
      href={props.goto}
      color={props.color}
    >
      {props.children}
    </Wrapper>
  );
};

export default Link;
