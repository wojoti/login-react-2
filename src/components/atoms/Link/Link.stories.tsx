import { ComponentMeta, ComponentStory } from "@storybook/react";

import Link from "./Link";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Atoms/Link",
  component: Link,
  decorators: [
    (Story) => (
      <div style={{ display: "flex", justifyContent: "center" }}>
        <Story />
      </div>
    ),
  ],
} as ComponentMeta<typeof Link>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Link> = (args) => <Link {...args} />;

export const Example = Template.bind({});
Example.args = {
  decoration: "none",
  color: "rgb(107 114 128)",
  children: "Forgot Password?",
};
