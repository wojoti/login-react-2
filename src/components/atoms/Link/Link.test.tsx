import { fireEvent, render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Link, { LinkProps } from "./Link";

const props: LinkProps = {
  children: "test link",
  onClick: () => {},
  testId: "test-link-id",
};

const modifiedProps: LinkProps = {
  ...props,
  decoration: "none",
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(<Link {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render matching link with text", () => {
  render(<Link {...props} />);
  const linkElement = screen.getByTestId("test-link-id");
  expect(linkElement).toBeInTheDocument();
  expect(linkElement).toHaveTextContent("test link");
});

test("should handle onClick event", () => {
  const handleClick = jest.fn();
  render(<Link {...props} onClick={handleClick} />);
  const linkElement = screen.getByTestId("test-link-id");
  fireEvent.click(linkElement);
  expect(handleClick).toHaveBeenCalledTimes(1);
});

test("should render link with modified style through props", () => {
  render(<Link {...modifiedProps} />);
  const linkElement = screen.getByTestId("test-link-id");
  expect(linkElement).toBeInTheDocument();
  expect(linkElement).toHaveStyle("text-decoration: none");
});
