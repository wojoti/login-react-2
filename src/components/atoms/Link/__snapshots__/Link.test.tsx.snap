// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<button
  className="sc-bcXHqe gRsFiG"
  data-testid="test-link-id"
  onClick={[Function]}
  type="button"
>
  test link
</button>
`;
